R/get.R

```diff
@@ -526,6 +526,76 @@ get_usage_static <- function(
   return(out)
 }
 
+#' Get usage information for deployed content
+#'
+#' @description
+
+#' Retrieve content hits for all available content on the server. Available
+#' content depends on the user whose API key is in use. Administrator accounts
+#' will receive data for all content on the server. Publishers will receive data
+#' for all content they own or collaborate on.
+#'
+#' If no date-times are provided, all usage data will be returned.
+
+#' @param client A `Connect` R6 client object.
+#' @param from Optional date-time (`POSIXct` or `POSIXlt`). Only
+#'   records after this time are returned. If not provided, records
+#'   are returned back to the first record available.
+#' @param to Optional date-time (`POSIXct` or `POSIXlt`). Only records
+#'   before this time are returned. If not provided, all records up to
+#'   the most recent are returned.
+#'
+#' @return A tibble with columns:
+#' * `id`: An identifier for the record.
+#' * `user_guid`: The GUID of logged-in visitors, NA for anonymous.
+#' * `content_guid`: The GUID of the content.
+#' * `timestamp`: The time of the hit as `POSIXct`.
+#' * `path`: The path of the hit. Not recorded for all content types.
+#' * `user_agent`: If available, the user agent string for the hit. Not
+#'   available for all records.
+#'
+#' @details
+#'
+#' The data returned by `get_usage()` includes all content types. For Shiny
+#' content, the `timestamp` indicates the *start* of the Shiny session.
+#' Additional fields for Shiny and non-Shiny are available respectively from
+#' `get_usage_shiny()` and `get_usage_static()`.
+#'
+#' When possible, however, we recommend using `get_usage()` over
+#' `get_usage_static()` or `get_usage_shiny()`, as it will perform better
+#' than those endpoints, which use pagination.
+#'
+#' @examples
+#' \dontrun{
+#' client <- connect()
+#'
+#' # Fetch the last 2 days of hits
+#' usage <- get_usage(client, from = Sys.Date() - 2, to = Sys.Date())
+#'
+#' # Fetch usage after a specified date
+#' usage <- get_usage(
+#'   client,
+#'   from = as.POSIXct("2025-05-02 12:40:00", tz = "UTC")
+#' )
+#'
+#' # Fetch all usage
+#' usage <- get_usage(client)
+#' }
+#'
+#' @export
+get_usage <- function(client, from = NULL, to = NULL) {
+  error_if_less_than(client$version, "2025.04.0")
+
+  usage_raw <- client$GET(
+    v1_url("instrumentation", "content", "hits"),
+    query = list(
+      from = make_timestamp(from),
+      to = make_timestamp(to)
+    )
+  )
+  usage <- parse_connectapi_typed(usage_raw, connectapi_ptypes$usage)
+  fast_unnest_character(usage, "data")
```

Comment:
Perhaps to consider in a followup: is there anywhere else this function should be used? For example, if you `include=owner` in GET `v1/content/`, that adds a list column right?